frontend/common/services/useUserEmail.ts

```diff
@@ -0,0 +1,52 @@
+import { Res } from 'common/types/responses'
+import { Req } from 'common/types/requests'
+import { service } from 'common/service'
+
+export const userEmailService = service
+  .enhanceEndpoints({ addTagTypes: ['UserEmail'] })
+  .injectEndpoints({
+    endpoints: (builder) => ({
+      updateUserEmail: builder.mutation<
+        Res['userEmail'],
+        Req['updateUserEmail']
+      >({
+        invalidatesTags: (res) => [
+          { id: 'LIST', type: 'UserEmail' },
+          { id: res?.id, type: 'UserEmail' },
+        ],
+        query: (query: Req['updateUserEmail']) => ({
+          body: query,
+          method: 'POST',
+          url: `auth/users/set_email/`,
+        }),
+      }),
+      // END OF ENDPOINTS
+    }),
+  })
+
+export async function updateUserEmail(
+  store: any,
+  data: Req['updateUserEmail'],
+  options?: Parameters<
+    typeof userEmailService.endpoints.updateUserEmail.initiate
+  >[1],
+) {
+  store.dispatch(
+    userEmailService.endpoints.updateUserEmail.initiate(data, options),
+  )
+  return Promise.all(
+    store.dispatch(userEmailService.util.getRunningQueriesThunk()),
```

Comment:
... this might be a silly question but what's a 'Thunk' 😄 ? `getRunningQueriesThunk`

Reply:
FE stuff

Reply:
Redux state works in a single direction in an immutable way, you do an action, it updates state which updates the ui

"Thunks" are basically side effects to actions, with thunks it's like
- Trigger action (e.g. update email)
- Thunk takes the action, does api stuff and generates an action to say it's loaded with the data
- Reducers take loaded action and update state